packages/react/src/TextInput/TextInput.tsx

```diff
@@ -10,17 +10,19 @@ import type { ForwardedRef, InputHTMLAttributes } from 'react'
 export type TextInputProps = {
   /** Whether the value fails a validation rule. */
   invalid?: boolean
+  /** The kind of data that the user should provide. */
+  type?: 'email' | 'password' | 'tel' | 'text' | 'url'
```

Comment:
Niels noticed this: we should test for these types in this components unit tests.

Reply:
I’ve added them.